bitcoin/src/crypto/schnorr.rs

```diff
@@ -176,6 +183,14 @@ impl TweakedKeyPair {
     pub fn to_inner(self) -> KeyPair {
         self.0
     }
+
+    /// Returns the [`TweakedPublicKey`] for this [`TweakedKeyPair`].
+    ///
+    /// This is equivalent to using [`TweakedPublicKey::from_keypair`].
+    #[inline]
+    pub fn public_key(&self) -> (TweakedPublicKey, Parity) {
```

Comment:
I don't think this should be called `public_key` if it also returns `Parity`. I suggest calling it `public_parts` (`parts` is common in Rust - `into_parts`, `Vec::from_raw_parts`... Even though this make make it look like we directly store the parts.)

Reply:
Cool, I'm down with that.

Reply:
It should be `TweakedPublicKey::parts_from_keypair()` also then, right?

Reply:
Oh, since that one is annoying to understand how about not returning parity in that case? As long as there's at least one method returning it we should be fine.

Reply:
Cool, done. Cheers.